sdk/src/wallet/operations/syncing/transactions.rs

```diff
@@ -151,66 +150,46 @@ where
-                                    // only reissue if inputs are still unspent
-                                    transactions_to_reissue.push(transaction);
-                                }
-                            }
+                        } else if input_got_spent {
+                            process_transaction_with_unknown_state(
+                                &wallet_data,
+                                transaction,
+                                &mut updated_transactions,
+                                &mut output_ids_to_unlock,
+                            )?;
                         }
                     }
                     Err(crate::client::Error::Node(crate::client::node_api::error::Error::NotFound(_))) => {
-                        // no need to reissue if one input got spent
                         if input_got_spent {
                             process_transaction_with_unknown_state(
                                 &wallet_data,
                                 transaction,
                                 &mut updated_transactions,
                                 &mut output_ids_to_unlock,
                             )?;
-                        } else {
-                            let time_now = unix_timestamp_now().as_millis();
-                            // Reissue if older than 30 seconds
-                            if transaction.timestamp + 30000 < time_now {
-                                // only reissue if inputs are still unspent
-                                transactions_to_reissue.push(transaction);
-                            }
                         }
                     }
                     Err(e) => return Err(e.into()),
                 }
+            } else if input_got_spent {
+                process_transaction_with_unknown_state(
+                    &wallet_data,
+                    transaction,
+                    &mut updated_transactions,
+                    &mut output_ids_to_unlock,
+                )?;
             } else {
-                // transaction wasn't submitted yet, so we have to send it again
-                // no need to reissue if one input got spent
-                if input_got_spent {
-                } else {
-                    // only reissue if inputs are still unspent
-                    transactions_to_reissue.push(transaction);
-                }
+                // Reissue if there was no block id yet, because then we also didn't burn any mana
+                log::debug!("[SYNC] reissue transaction {}", transaction.transaction_id);
+                let reissued_block = self
+                    .submit_signed_transaction(transaction.payload.clone(), None)
+                    .await?;
+                transaction.block_id.replace(reissued_block);
+                updated_transactions.push(transaction);
             }
         }
         drop(wallet_data);
 
-        for mut transaction in transactions_to_reissue {
```

Comment:
Any reason this was here before?

Reply:
Yes, it reissued txs that were still pending. This PR removes it since this requires additional Mana for the block which should be allotted first and it also might not work at all if there is a commitment context input, because it could be too old at this point

Reply:
okie